ygot/render.go

```diff
@@ -348,6 +351,25 @@ func TogNMINotifications(s GoStruct, ts int64, cfg GNMINotificationsConfig) ([]*
 	return msgs, nil
 }
 
+// TogNMISetRequest takes an input GoStruct and renders it to a single
+// SetRequest message. The configuration provided determines the path format
+// utilised, and the prefix to be included in the message if relevant.
+func TogNMISetRequest(s GoStruct, cfg GNMIConfig) (*gnmipb.SetRequest, error) {
+	var pfx *gnmiPath
+	if cfg.UsePathElem {
+		pfx = newPathElemGNMIPath(cfg.PathElemPrefix)
+	} else {
+		pfx = newStringSliceGNMIPath(cfg.StringSlicePrefix)
+	}
+
+	leaves := map[*path]interface{}{}
+	if err := findUpdatedLeaves(leaves, s, pfx); err != nil {
```

Comment:
In a gNMI `SetRequest` we can send a request that is JSON marshalled from the path that we have - so we can just  encode the GoStruct at a specified path using `ygot.Marshal7951`. This is likely to be preferable since it means that we have a structured document that can be read through rather than all `<path, value>` pairs it contains.